crime-means-assessment/src/test/java/uk/gov/justice/laa/crime/meansassessment/service/AssessmentSummaryServiceTest.java

```diff
@@ -0,0 +1,197 @@
+import static uk.gov.justice.laa.crime.meansassessment.staticdata.enums.WorkType.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class AssessmentSummaryServiceTest {
+
+    @Mock
+    private MaatCourtDataService maatCourtDataService;
+
+    @InjectMocks
+    private AssessmentSummaryService assessmentSummaryService;
+
+    private ApiCreateMeansAssessmentResponse meansAssessmentResponse;
+
+    @Before
+    public void setup() {
+        meansAssessmentResponse = TestModelDataBuilder.getCreateMeansAssessmentResponse(true);
+        meansAssessmentResponse.setRepId(1234);
+        meansAssessmentResponse.setAssessmentType(AssessmentType.INIT);
+        meansAssessmentResponse.setReviewType(ReviewType.ER);
+        meansAssessmentResponse.setInitialAssessmentDate(LocalDateTime.of(2021, 12, 20, 10, 0));
+    }
+
+    @Test
+    public void givenMeansAssessmentResponse_WhenGetAssessmentSummaryIsInvoked_ThenAssessmentSummaryShouldBeReturnedWithInitFinAssSummary() {
+        assessmentSummaryService.addAssessmentSummaryToMeansResponse(meansAssessmentResponse, null);
+
+        Assertions.assertThat(meansAssessmentResponse.getAssessmentSummary()).extracting("type")
+                .contains(Initial_Assessment)
+                .doesNotContain(Full_Means_Test, Passported, Hardship_Review_CrownCourt, Hardship_Review_Magistrate, IoJ_Appeal);
+        verify(maatCourtDataService).getPassportAssessmentFromRepId(1234, null);
+        verify(maatCourtDataService).getHardshipReviewFromRepId(1234, null);
+        verify(maatCourtDataService).getIOJAppealFromRepId(1234, null);
+    }
+
+    @Test
+    public void givenMeansAssessmentResponse_WhenGetAssessmentSummaryIsInvoked_ThenAssessmentSummaryShouldBeReturnedWithFullFinAssSummary() {
+        meansAssessmentResponse.setAssessmentType(AssessmentType.FULL);
+        assessmentSummaryService.addAssessmentSummaryToMeansResponse(meansAssessmentResponse, null);
+
+        Assertions.assertThat(meansAssessmentResponse.getAssessmentSummary()).extracting("type")
+                .contains(Full_Means_Test)
+                .doesNotContain(Initial_Assessment, Passported, Hardship_Review_CrownCourt, Hardship_Review_Magistrate, IoJ_Appeal);
+        verify(maatCourtDataService).getPassportAssessmentFromRepId(1234, null);
+        verify(maatCourtDataService).getHardshipReviewFromRepId(1234, null);
+        verify(maatCourtDataService).getIOJAppealFromRepId(1234, null);
+    }
+
+    @Test
+    public void givenMeansAssessmentResponseWithFullAssDate_WhenGetAssessmentSummaryIsInvoked_ThenAssessmentSummaryShouldBeReturnedWithFullFinAssSummary() {
+        meansAssessmentResponse.setAssessmentType(AssessmentType.FULL);
+        meansAssessmentResponse.setFullAssessmentDate(LocalDateTime.of(2022, 1, 20, 10, 0));
+        assessmentSummaryService.addAssessmentSummaryToMeansResponse(meansAssessmentResponse, null);
+
+        Assertions.assertThat(meansAssessmentResponse.getAssessmentSummary()).extracting("type")
```

Comment:
Is it worth adding in an additional assertion to check the correct assessment data has been set in the `getFinancialAssessmentSummary` logic? (The same would go for the previous two tests as well)

Reply:
yes definitely, I'll add asserts on date field